code/modules/surgery/organs/internal/cyberimp/augments_internal.dm

```diff
@@ -127,11 +131,31 @@
 		addtimer(CALLBACK(src, PROC_REF(clear_stuns)), stun_cap_amount, TIMER_UNIQUE|TIMER_OVERRIDE)
 
 /obj/item/organ/internal/cyberimp/brain/anti_stun/proc/clear_stuns()
-	if(owner || !(organ_flags & ORGAN_FAILING))
-		owner.SetStun(0)
-		owner.SetKnockdown(0)
-		owner.SetImmobilized(0)
-		owner.SetParalyzed(0)
+	if(isnull(owner) || (organ_flags & ORGAN_FAILING) || !COOLDOWN_FINISHED(src, implant_cooldown))
+		return
+	
+	owner.SetStun(0)
+	owner.SetKnockdown(0)
+	owner.SetImmobilized(0)
+	owner.SetParalyzed(0)
+	owner.setStaminaLoss(0)
+	addtimer(CALLBACK(owner, TYPE_PROC_REF(/mob/living, setStaminaLoss), 0), stun_resistance_time)
+	
+	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+	s.set_up(5, 1, src)
+	s.start()
```

Comment:
No new single letter vars please, prefer to rename the variable from `s` to `sparks` or something instead. Thank you for adding some type of this effect however